From gadiyar@ti.com  Wed Jul 15 06:34:11 2009
From: Anand Gadiyar <gadiyar@ti.com>
Date: Tue, 14 Jul 2009 16:41:00 +0530
Subject: USB: OMAP: OHCI: hc_driver's stop method should call ohci_stop
To: "linux-usb@vger.kernel.org" <linux-usb@vger.kernel.org>
Cc: "dbrownell@users.sourceforge.net" <dbrownell@users.sourceforge.net>, "linux-omap@vger.kernel.org" <linux-omap@vger.kernel.org>
Message-ID: <5A47E75E594F054BAF48C5E4FC4B92AB0305D662CF@dbde02.ent.ti.com>


From: Anand Gadiyar <gadiyar@ti.com>

OMAP: OHCI: hc_driver's stop method should call ohci_stop

Without this, the ohci-omap driver will not cleanup the debugfs
nodes when the driver is unloaded. So the next insmod will fail,
if CONFIG_DEBUG_FS and CONFIG_USB_DEBUG are both selected.

Reported-by: vikram pandita <vikram.pandita@ti.com>
Cc: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Anand Gadiyar <gadiyar@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-omap.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/host/ohci-omap.c
+++ b/drivers/usb/host/ohci-omap.c
@@ -282,6 +282,7 @@ static int ohci_omap_init(struct usb_hcd
 static void ohci_omap_stop(struct usb_hcd *hcd)
 {
 	dev_dbg(hcd->self.controller, "stopping USB Controller\n");
+	ohci_stop(hcd);
 	omap_ohci_clock_power(0);
 }
 
